### PR TITLE
Basic wires 1 reworked and updating logic, this design is likely to b…

### PR DESCRIPTION
**Pull request by anibor164**
This design will mostly likely will be trashed but there are some elements that are handy and the updated logic is closer to what well need.

### CODE REVIEW

File: default.htm [hunk 1]
```diff
@@ -68,11 +68,6 @@
 				margin-top: 6px;
 				/*TODO : better solution to spacing*/
 			}
-
-			.btn-label {
```
Comment (anibor164) [inline default.htm:72]: Removing this as it was a hack and these labels were using too much real estate

File: default.htm [hunk 2]
```diff
@@ -147,173 +142,156 @@ <h3>Keep Talking and Nobody Explodes</h3>
...
-									</div>
+									<input type="radio" class="btn-check" name="basicwires_red" id="basicwires_red1" autocomplete="off" value=1 onclick="basicwires_solve()">
+									<label class="btn btn-outline-danger" for="basicwires_red1">1</label>
 
-									<div class="btn-group-vertical  btn-group-sm" role="group">
-										<label class="btn btn-outline-primary btn-label">Blue</label>
+									<input type="radio" class="btn-check" name="basicwires_red" id="basicwires_red2" autocomplete="off" value=2 onclick="basicwires_solve()">
+									<label class="btn btn-outline-danger" for="basicwires_red2">2</label>
 
-										<input type="radio" class="btn-check" name="basicwires_blue" id="basicwires_blue0" autocomplete="off" value=0>
```
Comment (anibor164) [inline default.htm:184]: Removed the 0 option as these can be deselected now to count as 0

File: default.htm [hunk 3]
```diff
@@ -147,173 +142,156 @@ <h3>Keep Talking and Nobody Explodes</h3>
...
-										<label class="btn btn-outline-primary" for="basicwires_blue0">0</label>
+									<input type="radio" class="btn-check" name="basicwires_red" id="basicwires_red3" autocomplete="off" value=3 onclick="basicwires_solve()">
+									<label class="btn btn-outline-danger" for="basicwires_red3">3</label>
 
-										<input type="radio" class="btn-check" name="basicwires_blue" id="basicwires_blue1" autocomplete="off" value=1>
-										<label class="btn btn-outline-primary" for="basicwires_blue1">1</label>
+									<input type="radio" class="btn-check" name="basicwires_red" id="basicwires_red4" autocomplete="off" value=4 onclick="basicwires_solve()">
```
Comment (anibor164) [inline default.htm:164]: Added triggers to solve on each button

File: default.htm [hunk 4]
```diff
@@ -147,173 +142,156 @@ <h3>Keep Talking and Nobody Explodes</h3>
...
+									<label class="btn btn-outline-primary" for="basicwires_blue3">3</label>
 
-										<input type="radio" class="btn-check" name="basicwires_yellow" id="basicwires_yellow0" autocomplete="off" value=0>
-										<label class="btn btn-outline-warning" for="basicwires_yellow0">0</label>
+									<input type="radio" class="btn-check" name="basicwires_blue" id="basicwires_blue4" autocomplete="off" value=4 onclick="basicwires_solve()">
+									<label class="btn btn-outline-primary" for="basicwires_blue4">4</label>
 
-										<input type="radio" class="btn-check" name="basicwires_yellow" id="basicwires_yellow1" autocomplete="off" value=1>
-										<label class="btn btn-outline-warning" for="basicwires_yellow1">1</label>
+									<!--<input type="radio" class="btn-check" name="basicwires_blue" id="basicwires_blue5" autocomplete="off" value=5 onclick="basicwires_solve()">
```
Comment (anibor164) [inline default.htm:189]: Removed 5 and 6 buttons but likely the 4 isnt used either

File: default.htm [hunk 5]
```diff
@@ -425,175 +403,205 @@ <h3>Keep Talking and Nobody Explodes</h3>
...
+				document.getElementById("passwd_letters2").value.toLowerCase().split(''),
+				document.getElementById("passwd_letters3").value.toLowerCase().split(''),
+				document.getElementById("passwd_letters4").value.toLowerCase().split(''),
+				document.getElementById("passwd_letters5").value.toLowerCase().split('')
+			];
+			let matches = lookups.filter(lookupWord => {
+				let wordLetters = lookupWord.split('');
 
-				//wire buttons cannot be null, must have default and no deselect option. 
-				document.getElementById("basicwires_red0").checked = true;
```
Comment (anibor164) [inline default.htm:430]: Removed these default selections instead handling the null options as 0 wires

File: default.htm [hunk 6]
```diff
@@ -425,175 +403,205 @@ <h3>Keep Talking and Nobody Explodes</h3>
 				Array.from(document.querySelectorAll('#basicwires2_inputs .btn')).forEach(element => {
 					element.addEventListener('click', buttonDeselect);
 				});
+			document.getElementById("serial").style.display = "none";
```
Comment (anibor164) [inline default.htm:406]: Hide the serial buttons on load

File: default.htm [hunk 7]
```diff
@@ -425,175 +403,205 @@ <h3>Keep Talking and Nobody Explodes</h3>
...
+
+			});
+
+			document.getElementById("passwd_solution").innerHTML = matches.join('<br>');
+			console.log(matches);
+		}
+
+		function basicwires_solve(e) { //TODO:bug, no trigger for deselecting any button
+			
+			document.getElementById("basicwires_validation").innerHTML = "";
```
Comment (anibor164) [inline default.htm:445]: Added validation

File: default.htm [hunk 8]
```diff
@@ -425,175 +403,205 @@ <h3>Keep Talking and Nobody Explodes</h3>
...
+
+			let redWires = 0,
+				blueWires = 0,
+				yellowWires = 0,
+				blackWires = 0
+				whiteWires = 0,
+				serial = false,
+				lastwire = "";
+
+			if (document.querySelector('input[name="basicwires_last"]:checked') != null)
```
Comment (anibor164) [inline default.htm:456]: These can be cleaned up but the idea is there.

File: default.htm [hunk 9]
```diff
@@ -425,175 +403,205 @@ <h3>Keep Talking and Nobody Explodes</h3>
...
-								break;
-							}
-							if (blueWires > 1) {
-								result = last + " " + blue;
-								break;
-							}
+			var white = "white", yellow = "yellow", blue = "blue", red = "red", black = "black",
+				last = "last", first = "first", second = "2nd", fourth = "4th", third = "3rd";
+
+			if ((totalcount == 4 && redWires > 1) || (totalcount == 5 && lastwire == black) || (totalcount == 6 && yellowWires == 0)) {
```
Comment (anibor164) [inline default.htm:488]: Only options that need the serial input

File: default.htm [hunk 10]
```diff
@@ -425,175 +403,205 @@ <h3>Keep Talking and Nobody Explodes</h3>
...
+							result = `${last} ${blue}`;
+							break;
+						}
+						result = last;
+						break;
+					}
+				case 4:
+					{
+						if (redWires > 1 && serial) {
+							result = `${last} ${red}`;
```
Comment (anibor164) [inline default.htm:521]: Added literal strings :D thanks for showing me the ` trick here.

**Review by halojenproductions — APPROVED** [review]

**Review by halojenproductions — APPROVED** [review]

**Pull request closed without merging**